### ISSUE TITLE
Add tests to check that Service Configurations are not lost during upgrade/downgrade

### ISSUE THREAD

**Issue by Ebaneck** [scality/metalk8s#2293]
**Component**:

'tests', 'kubernetes'

**Why this is needed**:

We need to check and ensure that upon an upgrade or downgrade of a MetalK8s cluster, Service and Cluster configurations are not lost. 

**What should be done**:

In the CI, increment the value of a Service configuration (Dex replicas count) by editing the ConfigMap before an upgrade scenario.
Once the upgrade is done, during fast tests, check that the ConfigMap still holds the new values that were passed before cluster upgrade

**Implementation proposal** (strongly recommended):

TBD

**Test plan**:

**Comment by gdemonet** [comment]
> In the CI, increment the value of a Service configuration (Dex replicas count)

Would make sense to test another config than the one checked in #2292 

> Once the upgrade is done, during slow tests

Why **slow** tests? IMO, this kind of check is actually quite **fast**

**Comment by Ebaneck** [comment]
> > In the CI, increment the value of a Service configuration (Dex replicas count)
> 
> Would make sense to test another config than the one checked in #2292
> 
ok!
> > Once the upgrade is done, during slow tests
> 
> Why **slow** tests? IMO, this kind of check is actually quite **fast**

I see, i think i was influenced by when we ran slow and fast tests in the CI

Going to make the change in the ticket!

**Comment by gdemonet** [comment]
Slow and fast tests are really a matter of saving time. We'll run fast tests all the time, at all steps where we expect the current cluster to be fully functional, while try to run slow tests selectively (i.e. once after each of our scenarii: "install", "upgrade", etc.).

Now, this makes me think that orchestrating tests as described here will be harder than simply adding to what we currently have. As you noted in the issue, we currently run tests (either fast or slow), *before* or *after* doing the install/upgrade/whatever, assuming a fully working cluster for the whole duration. In this issue however, we describe a scenario for testing the effect (or in this case, non-effect) *of* running an upgrade:

```gherkin
Given I have a working cluster
And I have configured Dex to run 3 replicas
When I run an upgrade to version <X.Y.Z>
Then I still have Dex configured to run 3 replicas
(And Dex is still running 3 replicas)
```

Handling such scenarii is problematic, as upgrades are heavy, and sometimes almost irreversible, operations. We don't want to run one for each little test that ensures things keep working as we expect during and after the upgrade. Ideally, we'd be able to ensure all of these within a single upgrade test - practically, we can't, if we want to maintain sufficient isolation of our tests.

For this specific issue, designing a framework for running and aggregating said tests seems out of scope. However, we could focus on checking that all configurations (assuming they were left clean before and after the upgrade/downgrade/...) are set to the expected default values - _which could be a requirement/design limitation, actually: should we upgrade default values if these were left unchanged? cc @thomasdanan @NicolasT @TeddyAndrieux_. Running the tests before and after upgrade, in the expected versions, would then ensure the upgrade went well.

**Comment by Ebaneck** [comment]
First version was done by including a manual test plan
